code/game/turfs/change_turf.dm

```diff
@@ -97,8 +97,18 @@ GLOBAL_LIST_INIT(blacklisted_automated_baseturfs, typecacheof(list(
 
 	changing_turf = TRUE
 	qdel(src) //Just get the side effects and call Destroy
+	var/list/old_comp_lookup = comp_lookup?.Copy()
+	var/list/old_signal_procs = signal_procs?.Copy()
 	var/turf/W = new path(src)
 
+	// WARNING WARNING
+	// Turfs DO NOT lose their signals when they get replaced, REMEMBER THIS
+	// It's possible because turfs are fucked, and if you have one in a list and it's replaced with another one, the list ref points to the new turf
+	if(old_comp_lookup)
+		LAZYOR(W.comp_lookup, old_comp_lookup)
+	if(old_signal_procs)
+		LAZYOR(W.signal_procs, old_signal_procs)
```

Comment:
They're supposed to stay so why are you doing this as if they weren't?

Reply:
I do the lazyor in case some part of init registed for new signals, in order to prevent duplicates or failures to transfer/preserve

Reply:
With every new bit of information about turfs I learn I sink deeper into despair. Why do vis_contents transfer and nothing else.

Reply:
Discussed on discord, the real question here is "don't turf vars get copied when the turf is replaced?"
The answer is no, it seems that only happens with vis_contents for some insane reason. I hate byond